Spring-Boot/Spring-Boot-Academy/src/test/java/tw/waterball/judgegirl/springboot/academy/controllers/ExamControllerTest.java

```diff
@@ -599,27 +596,36 @@ void testGetStudentExamProgressOverview() throws Exception {
         assertEquals(secondQuestion.getProblemTitle(), otherProblem.getTitle());
     }
 
+    @DisplayName("Give an exam with three questions A,B,C and the problem related to B has been deleted,"
+            + "When get exam overview,"
+            + "question A should have problem title and problem be not archived and"
+            + "question B should be notFound and"
+            + "question C should have problem title and problem be archived .")
     @Test
     void testGetExamOverview() {
         final int QUOTA = 5;
         Date start = beforeCurrentTime(1, HOURS), end = afterCurrentTime(1, HOURS);
         ExamView exam = createExamAndGet(start, end, "exam");
         QuestionView q1 = createQuestionAndGet(new CreateQuestionUseCase.Request(exam.getId(), PROBLEM_ID, QUOTA, 50, 1));
         QuestionView q2 = createQuestionAndGet(new CreateQuestionUseCase.Request(exam.getId(), OTHER_PROBLEM_ID, QUOTA, 50, 2));
+        QuestionView q3 = createQuestionAndGet(new CreateQuestionUseCase.Request(exam.getId(), ARCHIVE_PROBLEM_ID, QUOTA, 50, 2));
+        problemServiceDriver.deleteProblem(q2.getProblemId());
 
         ExamOverview examOverview = getExamOverview(exam.getId());
-        ExamOverview.QuestionItem firstQuestion = examOverview.getQuestionById(new Question.Id(q1.examId, q1.problemId)).orElseThrow();
-        ExamOverview.QuestionItem secondQuestion = examOverview.getQuestionById(new Question.Id(q2.examId, q2.problemId)).orElseThrow();
+        var nonArchiveQuestion = examOverview.getQuestionById(new Question.Id(q1.examId, q1.problemId)).orElseThrow();
+        var notFoundQuestion = examOverview.getQuestionById(new Question.Id(q2.examId, q2.problemId)).orElseThrow();
+        var archivedQuestion = examOverview.getQuestionById(new Question.Id(q3.examId, q3.problemId)).orElseThrow();
```

Comment:
Your coding format should follow Given-When-Then pattern. Your code should look like this
```
/*
  Given Code Block
*/

/*
   When Code Block
*/

/*
   Then Code Block
*/
```
There won't be any changline in the code block unless it is necessary.